privacyidea/lib/token.py

```diff
@@ -2375,6 +2374,8 @@ def create_challenges_from_tokens(token_list, reply_dict, options=None):
                                              additional_tags=additional_tags)
             message_list.append(message)
             if challenge_created:
```

Comment:
the check if the challenge was created is here, the transactionid had already been assigned prior to this, causing the wrong behavior